app/src/views/ActiveSurgeDeployments/OngoingERUDeployments/index.tsx

```diff
@@ -1,9 +1,404 @@
+                    : undefined),
+            ) ?? [],
+        ).filter(isDefined);
+
+        const start = minSafe(startDateList);
+        const end = maxSafe(endDateList);
+
+        if (isNotDefined(start) || isNotDefined(end)) {
+            return undefined;
+        }
+
+        return {
+            start: new Date(start),
+            end: new Date(end),
+        };
+    }, [deployedERUResponse]);
+
+    const deployedEruDateRange = useMemo(() => {
+        if (
+            isNotDefined(deployedERUResponse)
+        || isNotDefined(deployedERUResponse.results)
+        || deployedERUResponse.results.length === 0
+        ) {
+            return { start: new Date(), end: new Date() };
+        }
+
+        const startDateList = deployedERUResponse.results.flatMap(
+            ({ active_erus }) => active_erus?.map(
+                (eru) => (isValidDate(eru.start_date)
+                    ? new Date(eru.start_date).getTime()
+                    : undefined),
+            ) ?? [],
+        ).filter(isDefined);
+
+        const endDateList = deployedERUResponse.results.flatMap(
+            ({ active_erus }) => active_erus?.map(
+                (eru) => (isValidDate(eru.end_date)
+                    ? new Date(eru.end_date).getTime()
+                    : undefined),
+            ) ?? [],
+        ).filter(isDefined);
+
+        const start = minSafe(startDateList);
+        const end = maxSafe(endDateList);
+
+        if (isNotDefined(start) || isNotDefined(end)) {
+            return { start: new Date(), end: new Date() };
+        }
+
+        return {
+            start: new Date(start),
+            end: new Date(end),
+        };
+    }, [deployedERUResponse]);
+
+    const dateRange = useMemo(() => {
+        if (
+            isNotDefined(deployedERUResponse)
+        || isNotDefined(deployedERUResponse.results)
+        || deployedERUResponse.results.length === 0
+        ) {
+            return undefined;
+        }
+
+        const startDateList = deployedERUResponse.results.flatMap(
+            ({ appeals }) => appeals?.map(
+                (appeal) => (isValidDate(appeal.start_date)
+                    ? new Date(appeal.start_date).getTime()
+                    : undefined),
+            ) ?? [],
+        ).filter(isDefined);
+
+        const endDateList = deployedERUResponse.results.flatMap(
+            ({ appeals }) => appeals?.map(
+                (appeal) => (isValidDate(appeal.end_date)
+                    ? new Date(appeal.end_date).getTime()
+                    : undefined),
+            ) ?? [],
+        ).filter(isDefined);
+
+        const start = minSafe([...startDateList, ...endDateList]);
+        const end = maxSafe([...startDateList, ...endDateList]);
+
+        if (isNotDefined(start) || isNotDefined(end)) {
+            return undefined;
+        }
+
+        return {
+            start: new Date(start),
+            end: new Date(end),
+        };
+    }, [deployedERUResponse]);
```

Comment:
There's a lot of code duplication here, we could create a new function that extracts the start and end dates from a given key.